data/actions/scripts/other/potions.lua

```diff
@@ -27,132 +72,90 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
 	if target == nil or not target:isPlayer() then
 		return true
 	end
-
+	
 	if player:getCondition(CONDITION_EXHAUST_HEAL) then
 		player:sendTextMessage(MESSAGE_STATUS_SMALL, Game.getReturnMessage(RETURNVALUE_YOUAREEXHAUSTED))
 		return true
 	end
 
-	local itemId = item:getId()
-	if itemId == antidotePot then
-		if not antidote:execute(target, numberToVariant(target:getId())) then
-			return false
-		end
-
+	local stats = potions[item:getId()]
+	if stats then
+		if (stats.vocations or stats.minLevel) and not player:getGroup():getAccess() then
+			if not stats.minLevel then
+				stats.minLevel = 1
+			end
+			if not stats.vocations then
+				stats.vocations = {}
+			end
+			
+			local vocationList, mvoc = {}, player:getVocation()
+			while mvoc do
+				table.insert(vocationList, mvoc:getId())
+				mvoc = mvoc:getDemotion()
+			end
+			-- Checking vocation, level
+			local isDifferentVocation = true
+			if #stats.vocations > 0 then
+				for k, v in ipairs(stats.vocations) do
+					if isInArray(vocationList, v) then
+						isDifferentVocation = false
+						break
+					end
+				end
+			else
+				isDifferentVocation = false
+			end
+			vocationList = nil
+			if isDifferentVocation or player:getLevel() < stats.minLevel then
+				local str, vocations = "", stats.vocations
+				if #vocations > 0 then
+					for i = 1, #vocations do
+						local voc = Vocation(vocations[i])
+						if i == #vocations then
+							str = str .. (#vocations ~= 1 and " and " or "") .. voc:getName():lower() .. "s"
+						else
+							str = str .. (i ~= 1 and ", " or "") .. voc:getName():lower() .. "s"
+						end
+					end
+				end
+				vocations = nil
+				if str == "" then
+					str = "players"
+				end
+				player:say("This potion can be consumed by ".. str .. " of level " .. stats.minLevel .. " or higher.", TALKTYPE_MONSTER_SAY)
+				return false
+			end
+		end
+		if stats.health then
+			local hmin, hmax = stats.health[1], stats.health[2]
```

Comment:
You could do the following: 
* potions = {[id] = {health = {**min = a**, **max = b**}}}
* potions[id].health.min, potions[id].health.max

and the same for mana points. This way it'll be clearer and you would not need to create more local variables.